### DISCUSSION TITLE
How to concatenate long logs (>16K are split)

### DISCUSSION THREAD

**Discussion by beckyjmcdabq** [fluent/fluentd#4220] [Q&A]
We occasionally have a log message greater than 16K that when scraped by fluentd and forwarded to Elasticsearch will appear as two separate documents.  This often happens with Java exceptions/stack traces being logged.

How can I configure fluentd to concatenate the separate logs into one log message before forwarding to Elasticsearch?

Google searching has pointed out using the "concat" filter but I'm struggling to find good documentation or an example of how I can get all the fields correct.  We are using the "tail" source to parse docker logs.  If someone can help with the configuration needed to support the long logs that get split, that would be most helpful.

**Comment by giakinh0823** [comment]
Same issue

**Reply by kenhys** [reply]
It seems this limitation comes from a logging driver.

https://github.com/moby/moby/issues/36777

If split log is well-formed, fluent-plugin-concat may help, but it is too optimistic.
or switching logging driver may help:

In the past, such a logging driver is created (I don't know it still works)
https://github.com/moby/moby/issues/25694#issuecomment-354138086
https://github.com/rchicoli/docker-log-elasticsearch/
